packages/amplify-util-mock/src/func/index.ts

```diff
@@ -1,67 +1,53 @@
-  const eventNameValidator = amplify.inputValidation({
-    operator: 'regex',
-    value: '^[a-zA-Z0-9/._-]+?\\.json$',
-    onErrorMsg: 'Provide a valid unix-like path to a .json file',
-    required: true,
-  });
-  let eventName: string = context.input.options ? context.input.options.event : undefined;
-  let promptForEvent = true;
-  if (eventName) {
-    const validatorOutput = eventNameValidator(eventName);
-    const isValid = typeof validatorOutput !== 'string';
-    if (!isValid) {
-      context.print.warning(validatorOutput);
+export async function start(context) {
+  const ampMeta = stateManager.getMeta();
+  let resourceName = context?.input?.subCommands?.[0];
+  if (!resourceName) {
+    const choices = _.keys(_.get(ampMeta, ['function'])).filter(resourceName => isMockable(context, resourceName).isMockable);
+    if (choices.length < 1) {
+      throw new Error('There are no mockable functions in the project. Use `amplify add function` to create one.');
+    } else if (choices.length == 1) {
+      resourceName = choices[0];
     } else {
-      promptForEvent = false;
+      const resourceNameQuestion = [
+        {
+          type: 'list',
+          name: 'resourceName',
+          message: 'Select the function to mock',
+          choices,
+        },
+      ];
+      ({ resourceName } = await inquirer.prompt<{ resourceName: string }>(resourceNameQuestion));
+    }
+  } else {
+    const mockable = isMockable(context, resourceName);
+    if (!mockable.isMockable) {
+      throw new Error(`Unable to mock ${resourceName}. ${mockable.reason}`);
     }
   }
 
-  if (promptForEvent) {
-    const resourceQuestions = [
-      {
-        type: 'input',
-        name: 'eventName',
-        message: `Provide the path to the event JSON object relative to ${resourcePath}`,
-        validate: eventNameValidator,
-        default: 'src/event.json',
-      },
-    ];
-    const resourceAnswers = await inquirer.prompt(resourceQuestions);
-    eventName = resourceAnswers.eventName as string;
-  }
-
-  const event = JSONUtilities.readJson(path.resolve(path.join(resourcePath, eventName)));
   const lambdaConfig = loadMinimalLambdaConfig(resourceName, { env: context.amplify.getEnvInfo().envName });
   if (!lambdaConfig || !lambdaConfig.handler) {
     throw new Error(`Could not parse handler for ${resourceName} from cloudformation file`);
   }
   const { allResources } = await context.amplify.getResourceStatus();
 
+  const event = await resolveEvent(context, resourceName);
   const envVars = hydrateAllEnvVars(allResources, lambdaConfig.environment);
+  context.print.warning('Ensuring latest function changes are built...');
+  await getBuilder(context, resourceName, BuildType.DEV)();
   const invoker = await getInvoker(context, { resourceName, handler: lambdaConfig.handler, envVars });
-  context.print.success('Starting execution...');
+  context.print.warning('Starting execution...');
```

Comment:
nit: using the warning text for these seems less than ideal. Perhaps another color like`context.print.blue` could be used?

Reply:
TIL we have a print.blue method haha. Good call, I'll use that